src/App.tsx

```diff
@@ -120,6 +122,32 @@ function App({ isEmbedded = false }: { isEmbedded?: boolean }) {
     embed,
   });
 
+  const shouldEnableZoomOutButton = useSelector(
+    canvasService,
+    (state) => canZoom(embed) && canZoomOut(state.context),
+  );
+
+  const shouldEnableZoomInButton = useSelector(
+    canvasService,
+    (state) => canZoom(embed) && canZoomIn(state.context),
+  );
+
+  const canShowWelcomeMessage = sourceState.hasTag('canShowWelcomeMessage');
+
+  const showControls = useMemo(
+    () => !embed?.isEmbedded || embed.controls,
+    [embed],
+  );
+
+  const showZoomButtonsInEmbed = useMemo(
+    () => !embed?.isEmbedded || (embed.controls && embed.zoom),
+    [embed],
+  );
+  const showPanButtonInEmbed = useMemo(
+    () => !embed?.isEmbedded || (embed.controls && embed.pan),
+    [embed],
+  );
```

Comment:
I would suspect that this can actually decrease performance than improve it ;p all of those are just booleans and computing them is trivial thus there is no need to memorize this stuff

Reply:
@Andarist this is in draft! No reviews yet please